config/daemon/ipv6.md

```diff
@@ -10,31 +10,110 @@ Before you can use IPv6 in Docker containers or swarm services, you need to
-> hosts.
+> **Note**
+>
+> - IPv6 support is experimental, use it with caution.
+> - IPv6 is only supported on Docker daemons running on Linux hosts.
 
-1.  Edit `/etc/docker/daemon.json`, set the `ipv6` key to `true` and the `fixed-cidr-v6` key
-    to your IPv6 subnet. In this example we are setting it to `2001:db8:1::/64`.
+## Daemon configuration
 
-    ```json
-    {
-      "ipv6": true,
-      "fixed-cidr-v6": "2001:db8:1::/64"
-    }
-    ```
+To enable IPv6, you must edit the Docker daemon configuration file located at
+`/etc/docker/daemon.json`. Configure the following parameters:
 
-    Save the file.
+```json
+{
+  "experimental": true,
+  "ipv6": true,
+  "ip6tables": true,
+  "fixed-cidr-v6": "2001:db8:1::/64",
+  ...
+}
+```
 
-2.  Reload the Docker configuration file.
+This configuration makes IPv6 networking function as you would expect it to.
+The `ipv6` and `fixed-cidr-v6` parameters are optional.
+They assign an IPv6 subnet to the default bridge network.
 
-    ```console
-    $ systemctl reload docker
-    ```
+After saving the configuration file, restart the Docker daemon for your
+changes to take effect:
 
-You can now create networks with the `--ipv6` flag and assign containers IPv6
-addresses using the `--ip6` flag.
+```console
+$ systemctl restart docker
+```
+
+Upon restart, the daemon assigns IPv6 addresses to containers connected to the
+default bridge network, and to user-defined networks configured with an IPv6 subnet.
+
+## Dynamic IPv6 subnet allocation
+
+If you don't explicitly configure subnets for user-defined networks,
+those networks use the default address pools of the daemon as a fallback.
+The default address pools are all IPv4 pools.
+
+To enable dynamic subnet allocation for user-defined IPv6 networks,
+you must manually configure address pools of the daemon to include:
+
+- The default IPv4 address pools
+- One or more IPv6 pools of your own
+
+The default address pool configuration is:
+
+```json
+{
+  "default-address-pools": [
+    { "base": "172.17.0.0/16", "size": 16 },
+    { "base": "172.18.0.0/16", "size": 16 },
+    { "base": "172.19.0.0/16", "size": 16 },
+    { "base": "172.20.0.0/14", "size": 16 },
+    { "base": "172.24.0.0/14", "size": 16 },
+    { "base": "172.28.0.0/14", "size": 16 },
+    { "base": "192.168.0.0/16", "size": 20 }
+  ]
+}
+```
+
+The following example shows a valid configuration with the default values and
+an IPv6 pool. The IPv6 pool in the example provides up to 256 IPv6 subnets of
+size `/112`, from an IPv6 supernet of prefix length `/104`. Each `/112`-sized
+subnet supports 65 536 IPv6 addresses.
+
+> **Note**
+>
+> Be aware that the following known limitations exist for IPv6 pools:
+>
+> - The `base` value for IPv6 needs a minimum prefix length of `/64`.
+>   This is due to an integer overflow in the Docker daemon.
+>   See [moby/moby#42801](https://github.com/moby/moby/issues/42801).
+> - The `base` supernet is split into subnets of `size`. Defining an excessive
```

Comment:
Do you want to remove the reference of `base` and `size` parameters? These were poorly documented in the past that anyone not familiar enough with networking (the likely audience) may have some trouble connecting the dots, especially since the syntax of the length portion vs the size value differs a bit.

I didn't feel there was a need to add the additional precision since we reference the actual Github issue with that information for anyone who'd need that information.

`24` is as I mentioned arbitrary as a limit, but the bigger issue is that no one really needs to define networks so large? Especially since IPv6 only networks aren't supported, if you're publishing ports for IPv4 to this many networks as well, I think it's safe to say the operator has the expertise to get this information and understand what excessive is.

For many users that aren't confident, they've been given an example of a sane pool configuration which provides 256 subnets of `2^16` IPv6 addresses, far more than the current private range advised for `default-address-pools` provides.

---

This is just personal opinion, but I would prefer to read docs that communicate what is appropriate to the bulk of the audience and their behaviour.

- Inexperienced users are typically going to copy the example (_and in this case probably use the reserved `2001:db8` since that'll work, and I've seen this already happening elsewhere_). It's also likely the advice third-party sites will parrot (_which I've also seen when prior IPv6 docs advised subnets of `/80`_).
- More curious readers may want to configure the settings, but in the past `base` and `size` were vague and lacked documentation on how they produced subnets. This snippet established the relation of these parameters to the appropriate jargon of `supernet` and `subnet` that they relate to, and how the networks are produced (_with an example below further iterating that connection_). Anyone concerned with what excess meant would visit the link for more information.
- This is technically a bug with the implementation, it's not meaningful information that will stay with the docs going forward. I would refrain from detailing the `24` "limit" on that basis, and so that we avoid another situation like with the `/80` subnet advice that was parroted around based on a statement there confusing users (_the MAC address behaviour still exists with `/80` sized subnets, but is undocumented_). Most users don't need so many networks created, but they'll likely do so redundantly if they feel it's "just to be safe".

Reply:
me and @akerouanton discussed to add a page dedicated to describing IP allocation and how that works, so we can cover this in the follow-up.